python/lsst/cp/verify/verifyStats.py

```diff
@@ -0,0 +1,510 @@
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+import math
+
+import lsst.afw.geom as afwGeom
+import lsst.afw.math as afwMath
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.pipe.base.connectionTypes as cT
+import lsst.meas.algorithms as measAlg
+
+from lsst.cp.pipe.cpCombine import VignetteExposure
+from lsst.pipe.tasks.repair import RepairTask
+from .utils import mergeStatDict
+
+__all__ = ['CpVerifyStatsConfig', 'CpVerifyStatsTask']
+
+
+class CpVerifyStatsConnections(pipeBase.PipelineTaskConnections,
+                               dimensions={"instrument", "exposure", "detector"},
+                               defaultTemplates={}):
+    inputExp = cT.Input(
+        name="postISRCCD",
+        doc="Input exposure to calculate statistics for.",
+        storageClass="Exposure",
+        dimensions=["instrument", "exposure", "detector"],
+    )
+    camera = cT.PrerequisiteInput(
+        name="camera",
+        storageClass="Camera",
+        doc="Input camera.",
+        dimensions=["instrument", ],
+        isCalibration=True,
+    )
+
+    outputStats = cT.Output(
+        name="detectorStats",
+        doc="Output statistics from cp_verify.",
+        storageClass="StructuredDataDict",
+        dimensions=["instrument", "exposure", "detector"],
+    )
+
+
+class CpVerifyStatsConfig(pipeBase.PipelineTaskConfig,
+                          pipelineConnections=CpVerifyStatsConnections):
+    """Configuration parameters for CpVerifyTask
+    """
+    maskNameList = pexConfig.ListField(
+        dtype=str,
+        doc="Mask list to exclude from statistics calculations.",
+        default=['DETECTED', 'BAD', 'NO_DATA'],
+    )
+    doVignette = pexConfig.Field(
+        dtype=bool,
+        doc="Mask vignetted regions?",
+        default=False,
+    )
+    doNormalize = pexConfig.Field(
+        dtype=bool,
+        doc="Normalize by exposure time?",
+        default=False,
+    )
+
+    doCR = pexConfig.Field(
+        dtype=bool,
+        doc="Run CR rejection?",
+        default=False,
+    )
+    repair = pexConfig.ConfigurableField(
+        target=RepairTask,
+        doc="Repair task to use.",
+    )
+    psfFwhm = pexConfig.Field(
+        dtype=float,
+        default=3.0,
+        doc="Repair PSF FWHM (pixels).",
+    )
+    psfSize = pexConfig.Field(
+        dtype=int,
+        default=21,
```

Comment:
Do we have any justification/mandate for the defaults?  Might the docstring provide some guidance as to if/when one might want to have different settings?

Reply:
These are the defaults inherited (via `cp_pipe`) from `pipe_drivers` `constructCalibs.py`.  This is run on bias, dark, and flat images, none of which have any real sources that would inform PSF options.  Thinking through the numbers, this should yield a narrow PSF (`\sigma ~ 1.27`) reaching out to approximately 8 sigma with this PSF size.  I don't think I have any strong guidance if it didn't remove all the obvious CRs, other than changing the PSF FWHM while keeping the size appropriate to prevent truncation.